x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityEsqlIT.java

```diff
@@ -306,30 +317,124 @@ public void testCrossClusterQuery() throws Exception {
                     | LIMIT 10"""));
                 assertOK(response);
 
-            }
-            // Check that authentication fails if we use a non-existent API key
-            updateClusterSettings(
-                randomBoolean()
-                    ? Settings.builder()
-                        .put("cluster.remote.invalid_remote.seeds", fulfillingCluster.getRemoteClusterServerEndpoint(0))
-                        .build()
-                    : Settings.builder()
-                        .put("cluster.remote.invalid_remote.mode", "proxy")
-                        .put("cluster.remote.invalid_remote.proxy_address", fulfillingCluster.getRemoteClusterServerEndpoint(0))
-                        .build()
-            );
-            for (String indices : List.of("my_remote_cluster:employees,employees", "my_remote_cluster:employees")) {
-                ResponseException error = expectThrows(ResponseException.class, () -> {
-                    var q = "FROM " + indices + "|  SORT emp_id DESC | LIMIT 10";
-                    performRequestWithLocalSearchUser(esqlRequest(q));
-                });
-                assertThat(error.getResponse().getStatusLine().getStatusCode(), equalTo(403));
-                assertThat(error.getResponse().getStatusLine().getStatusCode(), equalTo(401));
-                assertThat(error.getMessage(), containsString("unable to find apikey"));
+                Map<String, Object> responseAsMap = entityAsMap(response);
+                List<?> columns = (List<?>) responseAsMap.get("columns");
+                List<?> values = (List<?>) responseAsMap.get("values");
+                assertEquals(2, columns.size());
+                assertEquals(9, values.size());
+                List<String> flatList = values.stream()
+                    .flatMap(innerList -> innerList instanceof List ? ((List<String>) innerList).stream() : Stream.empty())
+                    .collect(Collectors.toList());
+                assertThat(
+                    flatList,
+                    containsInAnyOrder(
+                        "1",
+                        "2",
+                        "3",
+                        "4",
+                        "5",
+                        "6",
+                        "7",
+                        "8",
+                        "9",
+                        "engineering",
+                        "engineering",
+                        "engineering",
+                        "management",
+                        "sales",
+                        "sales",
+                        "marketing",
+                        "marketing",
+                        "support"
+                    )
+                );
             }
         }
     }
 
+    @SuppressWarnings("unchecked")
+    @AwaitsFix(bugUrl = "this trips ThreadPool.assertCurrentThreadPool(ThreadPool.Names.SEARCH_COORDINATION)")
```

Comment:
Do you need help on this?

Reply:
yes, i am not familiar enough with the threading model and think this is outside the concerns of the security model (e.g. probably best to address outside of this PR)

Reply:
I've opened https://github.com/elastic/elasticsearch/pull/107370.